homeassistant/components/bouncie/sensor.py

```diff
@@ -0,0 +1,181 @@
+"""Bouncie Sensors."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import BouncieDataUpdateCoordinator, const
+
+ATTRIBUTION = "Data provided by Bouncie"
+PARALLEL_UPDATES = 1
+
+SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="car-info",
+        icon="mdi:car",
+        name="Car Info",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="car-odometer",
+        icon="mdi:counter",
+        name="Car Odometer",
+        device_class=SensorDeviceClass.DISTANCE,
+    ),
+    SensorEntityDescription(
+        key="car-address",
+        icon="mdi:map-marker",
+        name="Car Address",
+    ),
+    SensorEntityDescription(
+        key="car-fuel",
+        icon="mdi:gas-station",
+        name="Car Fuel",
+        device_class=SensorDeviceClass.BATTERY,
+    ),
+    SensorEntityDescription(
+        key="car-speed",
+        icon="mdi:speedometer",
+        name="Car Speed",
+        device_class=SensorDeviceClass.SPEED,
+    ),
+    SensorEntityDescription(
+        key="car-mil",
+        icon="mdi:engine",
+        name="Car MIL",
+    ),
+    SensorEntityDescription(
+        key="car-battery",
+        icon="mdi:car-battery",
+        name="Car Battery",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Bouncie sensor entities based on a config entry."""
+    coordinator = hass.data[const.DOMAIN][config_entry.entry_id]
+    for vehicle_info in coordinator.data["vehicles"]:
+        async_add_entities(
+            BouncieSensor(coordinator, description, vehicle_info)
+            for description in SENSORS
+        )
+
+
+class BouncieSensor(CoordinatorEntity[BouncieDataUpdateCoordinator], SensorEntity):
+    """Bouncie sensor."""
+
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(
+        self,
+        coordinator: BouncieDataUpdateCoordinator,
+        description: SensorEntityDescription,
+        vehicle_info: dict,
+    ) -> None:
+        """Init the BouncieSensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._state = None
```

Comment:
The use of `_state` here looks wrong.
I think you should look at using a Callable/lambda on the entity description instead.

Reply:
Umm.. what do you mean? Any example that you can share?

Reply:
@epenet thoughts?

Reply:
Look for value_fn in the codebase